logging prototype

rpardini opened the pull request:
Of course not for merge

rpardini committed:
[NOPR] rpardini: config: configs and boards [NO PR]

- cloud-images, fastboot, more-like-ubuntu-cloud, output vmdk/qcow2 and many others
fastboot and oneplus msm898
- a LOT to fix in here still
- add QCOW2_RESIZE_AMOUNT for easy resize testing
- no longer x86 config
rpardini: renamed extension method
rpardini: logs
[rpardini] fix dual root rescue with extra space since it's so tiny
rpardini: a gazillion fixes for cloud-images, dual-root, etc pt 2
rpardini: a gazillion fixes for cloud-images, dual-root, etc
rpardini: fix mluc size
rpardini: yes, logs
rpardini: add package-list-utils, now under userpatches, where it belongs
rpardini: fix package-list-utils usage (non-reentrant)
rpardini: better cloud-like image
rpardini: build-machine-image
rpardini: my vmware lsmod
rpardini: cloud-init/mluc, fixes, do as late as possible; bring back armbian-config, it has neofetch
reformat rpardini only files - not oneplus
c-i/lvm-marker: allow configuring instance-id and better growmarker, now lvm-marker with both gpt and msdos partition support
[NOPR] oneplus5 and 5t
logging changes
[NOPR] WiP t95z using vim2 uboot and s96 dtb hijack
[NOPR] bye dual-root, hello rescue e2img, cloud-init motd, cleanups
- kernel major minor reference for oneplus
- move custom boards, kernel configs and families out of core into userpatches
- fix family include reference
- bump op5 to 5.12, last from JamiKettunen
- loose ends, 100% error handled now
- butterfly rebased
- bump rpardini to Jammy all around
- oneplus only 5.10
- fixes to rpardini extensions, war against hostapd has turned
- add usbquirks for uboot rtl9210
- [NOPR] build-machine-image: do not force BUILD_KSRC
- [NOPR] cleanup localmodconfig extension, works again
- cloud: block cloud+desktop builds, they make no sense
- [NOPR] squash enable zfs
- [NOPR] beginning of u-boot-menu extension [WiP]
- oneplus5: enable ZFS via dkms by enabling CONFIG_CRYPTO_DEFLATE which in turn enables CONFIG_ZLIB_DEFLATE
- AWS AMI extension, for use use with uefi-arm64-meta; serialcon; lsmods
- ami stuff, skip qcow2
- Etc/UTC as default timezone
- some gource fun
- rpardini: only do system-level checks if root
- rpardini: generic: auto cloud init URL.
- rpardini: generic cloud: introduce `CLOUD_INIT_RECIPE`
- rpardini: bump to KINETIC; disable CCACHE
- rpardini: t95z, limit uboot to 22.01
- rpardini: distcc config
- rpardini: t95z board: bump to 0.17 uboot SD from Khadas VIM2
- rpardini: warn when using bootenv
- rpardini: reenable CCACHE & disable usb quirks
- rpardini: resurrect the `oneplus5`/`msm8998` with new work from https://gitlab.com/msm8998-mainline/linux (6.1-rcX)
- rpardini: DON'T use my own firmware fork (rpardini/armbian-firmware) with libreElec firmware for Zero/Zero2
- remove empty oneplus5 msm8998-edge patches
- rpardini: configs: cloud-init: introduce `WIFI_SSID`, `WIFI_INTF` and `WIFI_PASS` to C-I wifi config via `cloud_init_modify_network_config()`
- rpardini: config: disable WIREGUARD; remove old/obsolete configs and readme's
- rpardini: configs: msm8998: edge: bump to 6.1.y
- aggregation.py -> rpardini configs
- rpardini t95z kernel DTS hack always enabled, no longer board-dependent
- config-rpardini : allow SKIP_EXTERNAL_TOOLCHAINS=no
- `phytium-d2000`, an uefi derivative separate-kernel 5.10 board, with working network driver
- rpardini: config: phytium lsmod with a lot of usbs plugged in
- rpardini: config: Phytium 4.19 vendor stuff; hammered config; split vendor patches by dir
- rpardini: config: enable ext kconfig-nocompress-nosign for all, in generic config
- rpardini: config: lsmod for beaglev from RichN's running machine
- rpardini: config: odroidn2 lsmod
- rpardini: configs: lsmod for tinkerboard-2 with some usb stuff, pci-e wifi, et al
- rpardini: configs: lsmod for 100% bare quartz64a (no pci-e, no usb)
- rpardini: config: update odroidhc4 lsmod
- rpardini: configs: remove old exts enablement
- rpardini: config: ksrc is no more
- rpardini: configs: remove old stuff
- rpardini: config: remove old stuff

lanefu reviewed (changes requested):
I didnt look at the code, therefore it must be improved

lanefu commented:
lots of progress!

rpardini committed:
------------------------ armbian-next END marker -----------------------------

- this contains .md with the git logs short and full for the armbian-next history pre-squash

rpardini committed:
------------------------ armbian-next END marker -----------------------------

- this contains .md with the git logs short and full for the armbian-next history pre-squash

rpardini committed:
[NOPR] rpardini userpatches: meson64's with SPIDEV and I2C EE M3 the hard way (c4,vim3,vim3l,zero,n2-plus,zero2) v6 (6.1.y/6.2.y)

- this adds spidev under spicc1, so it should be spidev1.0, but spicc0 not enabled, so it end the end becomes spidev0.0 -- good enough for now
- Add derived DTS with SPI (don't wanna mess with overlays for this)
- this counts on `drv-spi-spidev-remove-warnings` working
- rpardini: config: meson64: 6.1.y/edge: rebase SPI DTS
- rpardini: config: meson64: 6.1.y/current: rebase SPI DTS
- rpardini: config: meson64: 6.2.y/edge: rebase SPI DTS

rpardini committed:
[NOPR] rpardini: exts: custom extensions (mostly cloud-images) [NO PR]

- `java-build-machine` extension; includes GraalVM native-image for arm64
  - add zlib1g-dev for graal native builds.
- java-build-machine: bump to GraalVM 22.2.0
- rpardini exts don't fail if no resolved.conf (kinetic etc, probably underlying problem)
- rpardini ext cloudinit: force systemd-resolved install on kinetic
- rpardini: add extension `docker`, to pre-install `docker-ce` and such after adding gpg and apt
- rpardini: cloud: install wpasupplicant, always; cloud images can run on wifi-only SBCs too
- rpardini: hack/fix the fastboot extension to (barely) work again
- rpardini: cloud-init: introduce last-minute hook for network-config `cloud_init_modify_network_config()`
- rpardini: usbgadget, make more useful, work with modules; autoconfig NM for usb0-dhcp
- rpardini: exts: `cloud-metadata` ext, for real clouds, not SBC-fake-cloud.
- rpardini: ext: usbgadget fixes, shortcircuit build breaker fix
- aggregation.py -> rpardini exts
- rpardini exts uefi-rescue fix for CLOUD_INIT_INSTANCE_ID
- rpardini: exts: add extension for (previously-EXTRAWIFI) `rtl88x2bu` from https://github.com/cilynx/rtl88x2bu (ARCHIVE: extrawifi now works again)
- rpardini: exts: kconfig-nocompress-nosign.sh
- rpardini: exts: curb cloud-init debian logging
- rpardini: ext: kconfig-nocompress-nosign.sh: curb logging
- rpardini: exts: @TODO for e2img
- rpardini: ext: bm no longer has ksrc
- rpardini: exts: mluc/cloud: reorg packages; no use removing stuff .deb's will depend on later anyway
- rpardini: exts: drop old exts
- rpardini: exts: adapt kernel_headers_deb_rootfs to artifacts
- rpardini: cloud-init: curb logging
- rpardini: exts: fix fastboot shortcircuit

rpardini committed:
[kinetic] un-symlink `cli`/`main` `kinetic`->`jammy` packages; add `systemd-resolved` which was split off from main `systemd` package in Kinetic

rpardini committed:
`odroidc2`: u-boot: use minimal patchset for v22.01 u-boot

- symptoms:
  - no video display in u-boot
  - can't boot from SD (but does from eMMC)
- this has proven to solve nothing, but is still neat
- real problem is probably the FIP?

rpardini committed:
`risc-v`/`starfive`/`starfive2`/`d1`: all of RichNeese's Risc-V boards and families - squashed

- `risc-v`: rework RichNeese's boards, reduce families, move tweaks to boards
- `risc-v`/`starfive`: `CONFIG_MOTORCOMM_PHY=m` for the onboard Ethernet
- `risc-v`/`starfive`: use mainline kernel 6.1.y, with StarFive's rebased patches against `v6.1.5`
  - from https://github.com/starfive-tech/linux/commits/visionfive
  - contention point: `1022-soc-sifive-ccache-Add-StarFive-JH71x0-support.patch` which I merged half-assed, need review/fixes?
  - `risc-v`/`starfive`: update kernel config, sans changes
- `risc-v`/`starfive`: switch from `grub` to `extlinux`
- `risc-v`/`starfive`: new `starfive` family with their (vendor) kernel

Co-authored-by: Richard Neese <r.neese@gmail.com>

rpardini committed:
`media`: fixes for bad mbox/unneeded patches `media-edge` and `media-current` - thanks to @schwar3kat

- fix bad mbox for same old patch
- remove builddeb/mkdebian patches which are not needed in a-n

rpardini committed:
WiP: GHA Workflow generator delusion pt2

  - WiP: GHA Workflow generator delusion: experiment with aggregating kernel by git source+branch to group version resolution
  - WiP: GHA Workflow generator delusion: back to simple KernelPrep + Kernels + Images, no rootfs nor uboot
  - WiP: GHA Workflow generator delusion: fixes; fix dependency
  - WiP: GHA Workflow generator delusion: introduce `BaseAggregator::show_entries()` abstract
  - WiP: GHA Workflow generator delusion: half-abstract away the GHA Jobs, outputs, inputs, conditions, steps, needs; format
  - WiP: GHA Workflow generator delusion: split off Image from Input
- WiP: GHA Workflow generator delusion pt1.5
  - WiP: GHA Workflow generator delusion: split
  - WiP: GHA Workflow generator delusion: aggregate rootfs (half-done)
  - WiP: GHA Workflow generator delusion: coin flip for kernels up-to-date, outputs for skipped, use `always()`
  - WiP: GHA Workflow generator delusion: coin flip for kernels up-to-date
  - WiP: GHA Workflow generator delusion: prepare job for all kernels
  - WiP: GHA Workflow generator delusion: aggregator pt 2/3
  - WiP: GHA Workflow generator delusion: aggregator pt 1/3
  - WiP: GHA Workflow generator delusion: jobs with needs
  - WiP: GHA Workflow generator delusion: +split, add rootfs cli, add refs
  - WiP: GHA Workflow generator delusion: split

rpardini committed:
`sunxi/sunxi64`: disable some sunxi/sunxi64 "readme" patches that contain other patches -- for discussion

rpardini committed:
armbian-next: drop buildpkg code

rpardini committed:
`odroidm1`: enable working userland `fw_setenv`, so users can go back to Petitboot & otherwise tweak u-boot env from Armbian

rpardini committed:
`odroidhc4`: u-boot to 23.01; add HC4-SPI target preboot for USB; change HC4-SD target defconfig to new `hc4_sd`, similar to `c4`

- `odroidhc4`: pre-configure `fancontrol` so fans work right out of the box

rpardini committed:
`odroidm1`: bump to 6.2-rc3; look ma, no patches.

- pure mainline, https://github.com/torvalds/linux/commits/master/arch/arm64/boot/dts/rockchip/rk3568-odroid-m1.dts
- don't disable EXTRAWIFI

rpardini committed:
armbian-next: aggregation.py vs create-cache.sh: better hashing, much desktop

- aggregation.py:
  - only include non-removed pkgs in hash (do'h)
  - use a sorted unique of desktop+rootfs+debootstrap pkgs for pkg hash
  - make space for having apt sources/gpg keys in hash @TODO
  - write a hash-text variable for bash with info, for easy comparison later
  - refactor out bash_string_multiline()
- create-cache.sh:
  - don't armhf unless `ARMHF_ARCH=yes`
  - remove the uninstall stuff until we decide @TODO
  - don't show "free space of MOUNT" when there's only SDCARD
  - better logging for resolv.conf/machine-id shenanigans does in rootfs cache
  - drop dead code, squash some quoting issues
  - show the size of the zstd ball after compressing
  - don't even go to github armbian/cache nor cache.armbian.com for list of caches if SKIP_ARMBIAN_REPO=yes
  - remove mentions of BOARD or BRANCH, not relevant in rootfs cache

rpardini committed:
armbian-next: patching.py vs decent logging: better, but not there yet

- mostly downgrading stuff to .debug()
- some stats

rpardini committed:
armbian-next: nextify "Re-enabling support for Rpi3"

rpardini committed:
armbian-next: still fighting `tee` leaking under duress, turns out I _hadn't_ really won

- last resort, use lazy umount.
- include `lsof` in hostdeps, useful to debug these situations

rpardini committed:
armbian-next: still fighting `tee` leaking: now I think I won, again

- avoid using the "last resort, use lazy umount" by killing tee_pid _and descendants_ before hand. Works!
- introduce `get_descendants_of_pid_array()` using recursive trick to get all descendants
- do not `unset CURRENT_LOGFILE` -- during `logging_error_show_log()` (from SHOW_LOG=no days)
  - do `unset CURRENT_LOGFILE` immediately before processing logs -- although it magically survives. mysteries of traps in bash...
- `declare -g` instead of `export` for `start_logging_section()`
- don't `check_and_close_fd_13()` in `run_cleanup_handlers()`
- do run `check_and_close_fd_13()`, but only after archiving the old logs (in logging trap!)
- add a default error for unhandled TRAP's -- of which should be done
- fix reset of `global_tee_pid=0` in section-logging

rpardini committed:
armbian-next: somewhat-better handling/logging when using Docker (don't show two logfiles to inspect)

- do not show path to Docker run-host's logfile, unless the Docker run itself didn't produce logs
- introduce `global_final_exit_code` to allow to maintain/preserve Docker's exit code without triggering an error host-side

rpardini committed:
armbian-next: rename too-long main function to `prep_conf_main_build_single()`

- this commit is proof of OCD (old name didn't fit stacktrace space)

rpardini committed:
armbian-next: hopefully improved error logging; avoid logging errors up to 3x; subdued red stacktraces

- reorg `display_alert()` so more frequent levels are handled first
- improve some ANSI color usage, still a mess
- don't write failed commands stackstrace as warnings to screen -- just to file (yes it's redundant, except if in test)
- more useful / subdued stacktraces, to make actual error more obvious...

rpardini committed:
armbian-next: `patching`: less horrible patching log, pt 37

- don't show stats for patches that are bind applied blind (failed to parse/autogen)
- don't show Author, it's too long
- don't dump Python exceptions in log
- extra: expand Python imports

rpardini committed:
armbian-next: fix missing `BOARD_TYPE`

rpardini committed:
armbian-next: `git`: show "fetch completed" info, lest user thinks fetch is hanging while checking out

rpardini committed:
armbian-next: `patching`: less horrible patching log, pt 38: downgrade utf-8 mbox to log warning, mark as problem

rpardini committed:
`rk322x-box`: fix relative path reference to .bin in patch (a-n changed u-boot path on disk) - for @paolosabatino

rpardini committed:
armbian-next: fix ATF build under binutils 2.39+ via `TF_LDFLAGS=--no-warn-rwx-segment`

rpardini committed:
armbian-next: `kernel`: `git`: split via-ORAS and via-Bundle tangle

rpardini committed:
armbian-next: `kernel`: `git`: remove dead code, `GIT_PRE_FETCH_HOOK` / `checked_out_revision_mtime`

rpardini committed:
armbian-next: adaptive shallow/full Kernel git bare git (3gb -> 300mb download)

- allows using pre-shallow-ed bare trees, specific to the KERNEL_MAJOR_MINOR
- uses shallow by default if storage is "slow" (mmc), or free disk space < 32g (@TODO tweak this)
- if full tree already there keeps using it
- allow forcing decision with `KERNEL_GIT=full` or `KERNEL_GIT=shallow`
- countdown to confirm if decision based on storage type/free space and user is interactive
  - this has some of the worst copy ever written
- @TODO refactor the device type / free space code, reuse across codebase (3-4 other spots detect free space)
- @TODO GHA self-hosted (full) vs hosted (shallow) runners decision
- sprinkle with `wait_for_disk_sync()` around clean/download/extract
- sprinkle git with debugs, and an info between fetch and checkout (otherwise it seems "fetch" takes a long time)

rpardini committed:
armbian-next: `countdown`: split off file for countdown functions (countdown-to-abort and countdown-to-continue)

- introduce `countdown.sh`, move `traps.sh::exit_if_countdown_not_aborted()` there, introduce `exit_if_countdown_not_aborted()`

rpardini committed:
armbian-next: replace bare `sync` invocations by introducing `wait_for_disk_sync()`; sync early at start of build

- wait 10s for sync call; otherwise warn user to be patient
- wait_for_sync "reason for sync" everywhere
- add @TODO for /run/user/0 etc and XDG_RUNTIME_DIR in chroot

rpardini committed:
armbian-next: fix, don't try to get pid descendants on non-Linux (fixes Darwin cleanups)

rpardini committed:
armbian-next: darwin: try to get rid of `.DS_Store` sneaky bastards before launching Docker

rpardini committed:
armbian-next: back to cleaning downloaded bundles/gitballs after they've converted to bare tree & proven working

rpardini committed:
armbian-next: git: detect, and stop, when git bare trees vs worktrees have been shuffled around and can't be recovered

- hopefully this will never happen ;-)

rpardini committed:
armbian-next: git: disallow forcing impossible shallow/full; warn if it's the case

- you gotta wipe the worktree before you change it from full to shallow or vice-versa

rpardini committed:
armbian-next: docker: archive logfiles again; warn & greet testers when archiving many old logfiles

rpardini committed:
armbian-next: logging: new log file / summary nomenclature

rpardini committed:
armbian-next: bsp: use `prepare_temp_dir_in_workdir_and_schedule_cleanup()`; bring back `PACKAGES_SEARCH_ROOT_ABSOLUTE_DIRS`

rpardini committed:
armbian-next: firmware-deb.sh: prefix temp dir so it's "deb-firmware"

rpardini committed:
armbian-next: extensions: move 'extensions.sh' into library; replace manual source with init function

- wraps globals in `extension_manager_declare_globals()` -- and make them actually global
- whoever wrote this original code is insane, add copyright

igorpecovnik committed:
armbian-next: cli: add CLI alias command `rootfs-cache` for creating rootfs cache

rpardini committed:
armbian-next: fix ANSI logs, use reset at strategic places

rpardini committed:
armbian-next: `docker`: don't bomb if stdin is not a terminal (eg, in CI)

rpardini committed:
armbian-next: `docker`: pass `CI` env down to Docker (for GHA etc)

rpardini committed:
armbian-next: python/pip: solve crazy with `/usr/bin/pip3` by not using it (instead, `python3 -m pip`); don't mask errors

- "| cut" masked errors in multiple places
- include real pip3 version in python deps hash, all will be invalidated
- be less happy about hitting hash/cache, should happen often

rpardini committed:
armbian-next: `docker`: warn & countdown if we can't write to `cache/docker`; continue by using `.tmp/docker` if not aborted.

rpardini committed:
armbian-next: countdown: if to test `-t 0` (0: stdin, 1:stdout) for interactive-ness; also test in if-not-aborted version

rpardini committed:
armbian-next: interactive: info msg before (slow) gathering of all boards

rpardini committed:
armbian-next: logging: (very, very early beta) for uploading/sharing of ANSI log file online

rpardini committed:
armbian-next: kernel/logging: stop lying about LD/AR/INSTALL/SIGN/XZ make output

rpardini committed:
armbian-next: patching: accept b4's magic marker `git@z Thu Jan  1 00:00:00 1970` for mbox validations

rpardini committed:
armbian-next: cleanup, kill and add to-do's, remove dead code, turn down logging, squash future shortcircuit bugs

- killed `[[ $ROOTFS_TYPE != ext4 ]] && display_alert "Assuming ${BOARD} ${BRANCH} kernel supports ${ROOTFS_TYPE}" "" "wrn"`
  - which definitely didn't belong in rootfs
- disable usage of run_host_command_logged_long_running for kernel-make
- stop lying about long_running stuff being any different from non-long_running versions
- stop lying about 'set -e' when there's still a bunch of pipes for "pv" and stuff all around

rpardini committed:
armbian-next: mark `remove_all_trap_handlers()` deprecated, it never did anything

rpardini committed:
armbian-next: downloads.sh: some cleanup and add debugging to `get_urls()`

rpardini committed:
armbian-next: `apt`: expand local caching to include package lists, as well as .deb's

- `/var/cache/apt` in chroot was already cached;
- add `/var/lib/apt/lists`
- remove usage of namerefs, replace with global dict

rpardini committed:
armbian-next: `main`: split `default-build` into `build-packages`, `start-end`

- bear with me, this will make sense in the end

rpardini committed:
armbian-next: `rootfs`: split `create-cache` into `rootfs-create`

- bear with me, this will make sense in the end
- a `mount_chroot()` was ejected out of rootfs building code into main, for reuse's sake

rpardini committed:
armbian-next: `rootfs`: new CLI command for rootfs cache building; further refactorings

- _told ya it would all make sense eventually..._
- introduce new, real, non-alias `rootfs` command in `cli-rootfs` -- this _only_ builds rootfs, nothing else
- no more `main_default_build_single()`; instead `full_build_packages_rootfs_and_image()`
- introduce `do_with_default_build()` wrapper, for things that need host prepared + aggregation to run
- CLI-specific parts of `main_default_build_single()` moved to cli-build
- put prepare/cleanup pair of `prepare_rootfs_build_params_and_trap()` and `trap_handler_cleanup_rootfs_and_image()` into new `trap-rootfs.sh` (this needs further splitting for the image part)
- refactor `get_or_create_rootfs_cache_chroot_sdcard()` into `build_rootfs_only()`, `calculate_rootfs_cache_id()`
- update library with new files

rpardini committed:
armbian-next: `logging`: validate where not already in a logging section when trying to start a new one

- with a small exception for 'entrypoint' ;-)

rpardini committed:
armbian-next: `rootfs`: adapt to new reality after huge refactor

- make `tmpfs_estimated_size` and `available_physical_memory_mib` readonly and global and only ever set once
- remove subshell from `get_rootfs_cache_list()`, new version `get_rootfs_cache_list_into_array_variable()` for sanity
- warns if `ROOT_FS_CREATE_ONLY=yes` is set during cached build path
- add a `wait_for_disk_sync()` after restoring rootfs cache
- fix logging sections for `build_rootfs_only()` and `prepare_rootfs_build_params_and_trap()` in `build_rootfs_and_image()`
- remove the logging section from `get_or_create_rootfs_cache_chroot_sdcard()` call in `build_rootfs_only()` (which is now itself logged)
- @TODO: `deploy_qemu_binary_to_chroot()` is leaking a qemu bin in rootfs cache!

rpardini committed:
armbian-next: `aggregation`: dont' require or consider `BOARD` or `LINUXFAMILY` for aggregation anymore

- it was only used for "bsp-aggregation" of `rk3318-box` which I will come up with something extensions/hooks based later

rpardini committed:
armbian-next: `rootfs`: build rootfs from CLI, sans-BOARD, sans-FAMILY, etc -- rootfs are meant to be shared

- require ARCH/RELEASE for rootfs cli
- disallow BOARD/LINUXFAMILY/BOARDFAMILY being set for rootfs cli
- split `prep_conf_main_only_rootfs()` instead of `prep_conf_main_build_single()` for rootfs-only stuff
- run both conf and build without stdin for rootfs CLI (no interactive allowed)
- don't blindly set `ARCH=armhf` for no reason in main-config
- allow skipping family stuff in main-config, via `allow_no_family=yes` (used only in `prep_conf_main_only_rootfs()`)
- kill double loading of userpatches family, not needed/old code
- allow skipping kernel stuff via `skip_kernel=yes` in `config_pre_main()`  (used only in `prep_conf_main_only_rootfs()`)

rpardini committed:
armbian-next: `rootfs`: reset UID of produced rootfs tarball; announce what is doing and what has done

rpardini committed:
armbian-next: `rootfs`: converge `rootfs-cache` and `rootfs` together; exchange BOARD for ARCH

rpardini committed:
armbian-next: use unified methods for tmp dir handling all around (except ATF, which is special)

rpardini committed:
armbian-next: extensions: `nomod`: for when you really don't care about modules

rpardini committed:
armbian-next: we don't really need target `RELEASE` defined to install host deps (for now), don't freak out about it

rpardini committed:
armbian-next: kernel-debs: re-enable postinst debugging; fix bad builddeb port for zImage armhf (image_name == NAME_KERNEL)

rpardini committed:
@TODO for atftempdir

rpardini committed:
armbian-next: extrawifi: include `driver_uwe5622_allwinner()` in harness

rpardini committed:
armbian-next: extrawifi: include all `patch/misc/wireless*.patch` and `patch/misc/*wifi*.patch` in drivers hash calculation

- calculate hash of hash at once, not for each file

rpardini committed:
armbian-next: kernel-debs: force `arch/arm/boot/zImage` as vmlinuz for armhf

- we want a "Linux kernel ARM boot executable zImage (little-endian)" vmlinuz for armhf
- such is not deployed by default by kernel's "make install"

rpardini committed:
armbian-next: kernel-debs: squash some warnings; add `KERNEL_SRC_ARCH` to arch configs; don't warn about `NAME_KERNEL`

rpardini committed:
armbian-next: wait_for_disk_sync(): warn only if more than 30s fsync

rpardini committed:
`meson64`: edge/6.2: spi-nor: add support for XT25Q64 in addition to XT25F128B

rpardini committed:
`meson64`: edge/6.2: bump to `6.2-rc5`

rpardini committed:
`meson64`: edge/6.2: rebase onto `v6.2-rc5`

rpardini committed:
armbian-next: binfmt: make sure we have all `binfmt`'s setup & enabled for `rootfs` build (v2)

rpardini committed:
armbian-next: `docker`: don't spew errors about sharing cache if cache doesn't even exist

rpardini committed:
armbian-next: GHA & rootfs adventures, pt 1

- rootfs: create readonly global `rootfs_cache_id` (type+hash+date/version)
- add "oras-upload" CLI command (dumb, no retries, single target/single source)
- docker: mount-bind for & re-pass envs `GITHUB_OUTPUT` and `GITHUB_STEP_SUMMARY` down to Docker
- introduce `github-actions.sh::github_actions_add_output()`
- during logs cleanup, dump the Markdown log into GITHUB_STEP_SUMMARY if it is available

igorpecovnik committed:
armbian-next: rootfs: introduce `ROOTFS_COMPRESSION_RATIO`, default 5 for general use, default 15 for `rootfs` CLI; fix pv | tar bug; remove GPG signing

- Remove PGP signing from the script as its done in GHA
- Also remove generating .current control file as its obsolete in this design

Co-authored-by: Ricardo Pardini <ricardo@pardini.net>

rpardini committed:
armbian-next: aggregation: quick & dirty "potential paths" in Markdown

rpardini committed:
armbian-next: cli: `kernel` cli forces `KERNEL_CONFIGURE=no`; @TODO this should only "default to", not force. for proper kernel cli, later

rpardini committed:
armbian-next: logging: pure ASCII .log; included in Markdown for GHA

rpardini committed:
armbian-next: docker/gha: pass down a bunch of GHA specific vars (list unconfirmed)

- docker: curb logging about stdin

rpardini committed:
armbian-next: rootfs: curb logging; GHA outputs `rootfs_cache_id_version`, `rootfs_cache_id` and `rootfs_out_filename_relative`

rpardini committed:
`meson64`: edge/6.2: arm64: dts: amlogic: meson-sm1-odroid-hc4: fix active fan thermal trip

rpardini committed:
armbian-next: patching: rewrite: don't gag on bad "/dev/null.c" rename patches; archeology: `--find-copies-harder`; try to get GHA to run GH pages

- try to convince GHA to actually run my kernel patching summary workflow on push

rpardini committed:
`rockchip64`: current/5.15: fix bad unidiff/lack of mbox of wifi-hf-lps170-bl602.patch & makefile

rpardini committed:
armbian-next: chroot_sdcard_apt_get_install_dry_run(): run again if first try fails, with more logging

rpardini committed:
armbian-next: `docker`: passthrough of the SSH agent, via Docker magic on Darwin; enable for `docker-shell` and `kernel-patches-to-git`

- don't check/verify/update SSH-related hostkey etc for git tree push

rpardini committed:
`rockchip64`: current/5.15: remove IPX revert patch that's already in drivers now

rpardini committed:
armbian-next: `docker`: include `${SRC}/.git` static via Dockerfile, for `kernel-patches-to-git` / archeology; use fast archeology

- include FAST_ARCHEOLOGY in call to Python, default yes; sync kernel & u-boot call

rpardini committed:
armbian-next: all chroots: mount `/var/tmp` as tmpfs; create dir `/run/user/0` and mount as tmpfs @TODO cleanups

rpardini committed:
armbian-next: docker: pass `ARMBIAN_ENABLE_CALL_TRACING=yes` via env, and only if `DOCKER_ARMBIAN_ENABLE_CALL_TRACING=yes`

rpardini committed:
armbian-next: compress-checksum: move config to config-prepare; drop weight; kill warnings

rpardini committed:
armbian-next: repeat build opts: fix missing BOARD, show both after config (for logfile) & after build

rpardini committed:
armbian-next: bump logging archive greet to 14 files

rpardini committed:
armbian-next: fix: kernel-git-oras: make sure `cache_git_bare_dir` exists before trying to decide which media it's on

rpardini committed:
armbian-next: aggregate_packages out of `main_default_start_build()`; introduce `assert_requires_aggregation()`

rpardini committed:
armbian-next: git: `fetch_from_repo()`: add fetched_revision(_ts) as well as checked_out_revision(_ts); introduce param `do_checkout=no` to only fetch, not checkout

rpardini committed:
armbian-next: firmware-deb: reworked: Provides/Conflicts instead of Replaces; `cp` -> `git archive`; new firmware CLI; FULL version .deb out of tmpfs

rpardini committed:
armbian-next: cli-build: don't show two `BOARD=` repeat option

rpardini committed:
armbian-next: unify 2 methods for config: `prep_conf_main_minimal_ni()` (rootfs/firmware/kernel) and `prep_conf_main_build_single()` (full-build, possibly interactive)

rpardini committed:
armbian-next: curb some logging for git non-checkouts

rpardini committed:
armbian-next: prepare-host:  don't manage ACNG in prepare if not asked to

rpardini committed:
armbian-next: add Debian Ports keyring hostdep to strap Debian riscv64 from ports

- rpardini: actually use ports keyring for Debian Risc-V
- `risc-v`: correctly include `debian-ports-archive-keyring` in Risc-V CLI packages
  - include in debootstrap list, via `config/optional/architectures/riscv64/_config/cli/_all_distributions/main/packages`
  - can't have two `--include` in debootstrap call
- riscv64: move `debian-ports-archive-keyring` from main to debootstrap; also only for `sid`

rpardini committed:
armbian-next: ext: `grub-riscv64`: syncs with original version, fixes, nextify

rpardini committed:
armbian-next: ext: `grub-sbc-media`: syncs with original version, fixes, nextify

rpardini committed:
armbian-next: docker: fix: not all `find`'s are made equal, back to `-mtime +1` (full 24-hour period needed for match)

rpardini committed:
armbian-next: interactive: review `-t 0` instead of `-t 1`; move non-interactive `check_basic_host()` from prepare-host to config-prepare, for consistency

rpardini committed:
armbian-next: move aggregation-had-run check to more strategic places (pre-bsp, pre-rootfs) instead of at the beginning of full_build_packages_rootfs_and_image()

rpardini committed:
armbian-next: docker: split `docker_cli_build_dockerfile()` into `docker_cli_prepare_dockerfile()`; introduce `FAST_DOCKER`

- armbian-next: docker: fix hostrelease crazy

rpardini committed:
armbian-next: debs: introduce `shellcheck` for `DEBIAN/postinst` and others, before dpkg build

rpardini committed:
armbian-next: git: refactor `git_parse_ref()` out of `fetch_from_repo()` for reuse

rpardini committed:
armbian-next: config: twist `prep_conf_main_minimal_ni()` to receive `use_board=`

rpardini committed:
armbian-next: artifacts: introduce kernel/u-boot artifacts; git ref2info; "memoizer"; some hashing

- artifacts: u-boot/kernel - pt7 - adapt legacy/artifact versions; use common `capture_rename_legacy_debs_into_artifacts()`
- artifacts: u-boot - pt6: add artifact for u-boot
  - use artifact version / reason in actual u-boot .deb if present
- artifacts: kernel - pt5: tune kernel version, refactor
- artifacts: kernel - pt4: squash unrelated bugs that show up; move `prepare_compilation_vars()` to default build
- artifacts: kernel - pt3: drivers+patches+.config hashing
  - split file hashing function from drivers-harness; fix it so filenames are relative and sorted; sort from ${SRC}, always
  - aplit prepare_kernel_config_core_or_userpatches() from `kernel_config_initialize()`
- artifacts: kernel - pt2:  memoizing git ref2info
- artifacts: kernel - pt1:  versioning

rpardini committed:
`v6.2-rc7`: bump `edge` to `v6.2-rc7` for UEFI / meson64 / rk3568-odroid / rockchip-rk3588

rpardini committed:
armbian-next: do-not-fix docker-asking-for-docker error by adding a "tip"

rpardini committed:
armbian-next: cli: *do not* allow running as root on terminal; make `PREFER_DOCKER=no` actually work

- do allow root if non-interactive

rpardini committed:
armbian-next: cli: try (and fail, probably...) to be helpful with half-working Docker installs

rpardini committed:
armbian-next: logging: detect `TERM=alacritty` and use normal instead of gray

rpardini committed:
armbian-next: fix: for the 1000th time had broken `KERNEL_ONLY=yes`; curb some warnings

rpardini committed:
armbian-next: introduce `assert_prepared_host()` and assert before using any Python, to un-mask any further fallacies

rpardini committed:
armbian-next: take a breath: config might mark aggregation required; aggregation requires prepare_host, and thus runs in default-build

rpardini committed:
armbian-next: logging: ansi-haste sharing beta 2 (now with URL)

rpardini committed:
armbian-next: docker: fix Dockerfile generation; don't output logs info during Dockerfile build

rpardini committed:
armbian-next: u-boot: downgrade more errors to warnings, so we can build some legacy u-boot's

rpardini committed:
`sun50iw9`: half-fixes to get going under armbian-next (@TODO needs review/nextify/confirm busybox is not needed)

rpardini committed:
armbian-next: cli: u-boot alias: don't stop to interactively ask for kernel configuration during u-boot build

rpardini committed:
armbian-next: call-stack-analyze.py: detect missing function_to_file and stop properly (half-finished traces?)

rpardini committed:
armbian-next: json-info: cleanup info/json/csv/opensearch, add logging, add some very basic instructions; add Kibana dashboards & docker-compose to bring it OS+Kibana up

rpardini committed:
armbian-next: shellcheck: update, cleanup script, invent some new severities

rpardini committed:
armbian-next: shellcheck: squash a bunch of 'critical' warnings

rpardini committed:
armbian-next: shellcheck: raise the bar, stop excluding "SC2155 Declare and assign separately to avoid masking return values"

rpardini committed:
armbian-next: squash some shellcheck warnings; don't `local` and assign subshell value in same statement, that masks errors (`declare`/`local` almost-always works)

rpardini committed:
armbian-next: shellcheck: raise the bar, stop excluding "SC2154 is referenced but not assigned"

rpardini committed:
armbian-next: shellcheck: squash more shellcheck warnings; some long-lost variables being used; some unset's that are better reset's, etc

rpardini committed:
armbian-next: `docker-shell`: provide SSH and forward ssh auth socket to `docker-shell`

rpardini committed:
armbian-next: docker: pass `CARD_DEVICE` to Docker via `--device`

rpardini committed:
armbian-next: u-boot: mark bare repo as safe before using it (@TODO `git_ensure_safe_directory()` stuff, again)

rpardini committed:
armbian-next: swap github in as default u-boot mirror; allow `denx` to use upstream's gitlab

- @TODO still need to change mirror swap to usage instead of config

rpardini committed:
armbian-next: git ref2info: dereference annotated tags to get "real" commit SHA1; fail if curl failed; validate parse_makefile_version()

amazingfate committed:
`rockchip64`: update bl31 and ddr blobs of rk3568 [cherry-pick]

rpardini committed:
armbian-next: hash/cache: remove some cruft, some better logging

rpardini committed:
armbian-next: docker: pass down git info via env var, since Docker doesn't get `${SRC}/.git`, yet we need that info in the Docker logs

rpardini committed:
armbian-next: u-boot: fix: check gcc version before trying to downgrade certain errors to warnings

Kreyren committed:
boards/README: Document SKIP_ARMBIAN_REPO [cherry-pick]

See ./lib/functions/rootfs/distro-specific.sh

rpardini committed:
armbian-next: tmpfs: use `size=99%` for all tmpfs's; (rootfs/WORKDIR/chroot); thanks Igor for suggestion

rpardini committed:
armbian-next: armbian-kernel: default .config options set for all kernels

amazingfate committed:
update rk35xx legacy kernel to 5.10

amazingfate committed:
add radxa e25

rpardini committed:
armbian-next: git: backport `safe.directory` method from master (just add `*`, once) -- thanks Igor

rpardini committed:
armbian-next: logging: introduce `SHARE_LOG=yes` to auto-curl the logs to paste

rpardini committed:
armbian-next: logging: introduce `RAW_LOG=yes` to export .tar with raw unprocessed logs (for development)

rpardini committed:
armbian-next: tooling: include ORAS tools in Dockerfile; use them if available instead of downloading

rpardini committed:
armbian-next: tooling: introduce helper/downloader for shellcheck; include in Dockerfile

rpardini committed:
armbian-next: rootfs: include hash of rootfs bash code in the `packages_hash`/`rootfs_cache_id`

- don't use dash separator for `B`ash hash in `packages_hash`/`rootfs_cache_id`

rpardini committed:
armbian-next: python/pip: include pip pkgs in Dockerfile; use them if available instead of downloading

SteeManMI committed:
`meson64`: `current`: Enable MGLRU on meson64-current [cherry-pick]

Per PR#4697 this was enabled first on meson64-edge to test.
I am now following up to enable on meson64-current.

amazingfate committed:
enable serial output on rock-3a and radxa-e25

Kreyren committed:
`bookworm`: has new/separate repos for non-free-firmware  (#4801) [cherry-pick]

- "Debian moved the non-free firmware into it's own sub-repository"

rpardini committed:
armbian-next: artifacts: firmware/rootfs and actual usage of artifacts during image build

- artifacts: introduce `ARTIFACT_IGNORE_CACHE=yes`
- artifacts: introduce `DONT_BUILD_ARTIFACTS`, list of artifacts that if not found cached, fail the build
- kernel_package_source() is no more
- a long dissertation about kernels, families, and the universe
- artifacts: actually use rootfs artifact for image build
- artifacts: detangle via artifact_base_dir
- artifacts: rootfs: use folders in artifact_name; include cache_type
- artifacts: some cleanups / validations
- rootfs artifact; drop old rootfs cli
- artifacts: new CLI shortcuts; remove old firmware CLI
- artifacts: full firmware & usage
- use firmware artifacts in image build and install its debs
- kernel artifact sans legacy; use tmpdir for .deb target for all packages
- legacy artifact versions is no more; pack/unpack now in common obtain;
- artifacts: uboot: cleanup legacy renaming, use artifact version directly
- artifacts: add firmware (small) artifact
- deploy uboot to loop from artifact; allow tty to artifact; todos for cleaning
- fixes, kernel dtb/headers conditional; remove `.git` from Makefile url; use mapfile for finding files to hash
  - completely remove KERNEL_HAS_WORKING_HEADERS_FULL_SOURCE and `kernel_package_callback_linux_headers_full_source()`
  - don't use realpath for artifact_file_relative
  - curb some warnings
  - fix: only install headers & dtbs if such exist
- kernel .config hook modification hash versioning
- OCI_TARGET_BASE vs per-artifact defaults; only deploy to remote from CLI with OTB
- artifact revolver & installing into image
  - add artifact_map_packages and artifact_map_debs dicts
  - revolver accumulates all info
  - REPOSITORY_INSTALL is no more (for uboot/kernel, later others)
  - rename `VER` to `IMAGE_INSTALLED_KERNEL_VERSION`

amazingfate committed:
use sdk branch for rk35xx legacy

amazingfate committed:
fix radxa e25 uboot and kernel boot

rpardini committed:
armbian-next: rootfs: fix: too many bash files included in rootfs `B` hash, reduce to only what is used to produce rootfs tarball

- otherwise we get rootfs cache misses when working distro-agnostic and others, which are not part of rootfs code

amazingfate committed:
remove mmc alias for rock3a and radxa-e25

rpardini committed:
armbian-next: binfmts: suppress error message when trying to emulate armhf on arm64

rpardini committed:
rpardini: configs: amlogic 5.4 kernel for vim3l - lukefor's defconfig + config updated + wifi usb stuff - ttyS0 - bootscript - wifi stuff - full firmware + instructions

rpardini committed:
armbian-next: git/ref2info: fixes for using `commit:<sha1>`

rpardini committed:
armbian-next: kernel: pkg: dtb: show 2 levels of dtbs in tree

rpardini committed:
armbian-next: armbian-kernel: force `CONFIG_LOCALVERSION_AUTO=n` and `DEBUG_INFO=n`

rpardini committed:
armbian-next: bsp-cli: fix shellcheck warning

Kreyren committed:
distro-agnostic: `dropbear-initramfs`: make sure directory exists [cherry-pick]

rpardini committed:
armbian-next: hash-files: if no files to hash, return 16 zeros, instead of hanging forever waiting for stdin

rpardini committed:
armbian-next: git-ref2info: add support for `gitlab.com`; exit_with_error if SHA1 can't be found

rpardini committed:
rpardini: configs: crazy quartz64a non-stuff

rpardini committed:
armbian-next: remove all traces of Vagrant and FEL boot

Commit:
sid: add unstable repo

ashh87 committed:
`meson64`: `current` + `edge`: bring back fixed version of `odroid-reboot` driver (Fix reboot on odroid C4 when using UHS-II SD cards) [cherry pick]

rpardini committed:
armbian-next: docker: show $SECONDS in Docker dash logs

rpardini committed:
armbian-next: aggregation: don't include `packages.additional` if `BUILD_MINIMAL=yes`

rpardini committed:
armbian-next: chroot-helpers: if unable to mount `devpts`, use a mount bind from host

Co-authored-by: Paolo Sabatino <paolo.sabatino@gmail.com>

igorpecovnik committed:
Clean / unify gnome desktop packages [cherry-pick]

igorpecovnik committed:
Clean Mate desktops [cherry-pick]

rpardini committed:
armbian-next: chroot-helpers: mount_chroot()/umount_chroot(): add warning with stacktrace if called sans logging section

rpardini committed:
armbian-next: build_rootfs_and_image(): log correctly mount_chroot()/umount_chroot() calls

rpardini committed:
armbian-next: artifacts (all): curb logging, use  more descriptive git source var names

rpardini committed:
armbian-next: cleaning: deprecate `CLEAN_LEVEL=debs` -- doesn't do anything, just shows what would be cleaned

rpardini committed:
armbian-next: debug_dict()/debug_var() only if debugging

rpardini committed:
armbian-next: remove references to stuff that doesn't exist or isn't hit by default anymore

rpardini committed:
split BOARD `bananapim2pro` from `bananapim5`, and force use of its new DTB; only for 6.2+, so edge only; use m5 FIP for m2pro

rpardini committed:
armbian-next: docker: curb logging about OCI_TARGET_BASE

rpardini committed:
armbian-next: oras: add 3/5 retries to oras pull/push

rpardini committed:
armbian-next: post_build_image: run hook first, then write to CARD_DEVICE, then compress, in that order

- compress *all* present output images, not only .img
- remove 7zip compression & hostdep
- squash a few shortcircuits (yeah, this late in the game...)

rpardini committed:
armbian-next: compression: fast xz compression for images

rpardini committed:
armbian-next: rootfs artifact: don't try grouping OCI target, it does not really work

rpardini committed:
armbian-next: memoize: expire all memoized files after 1 hour

rpardini committed:
`qemu-uboot-xxx`: fix missed rootfs cache due to pkg removals

rpardini committed:
armbian-next: oci-oras: don't capture our own debugs as part of the manifest (d'oh -- thanks RichN)

rpardini committed:
armbian-next: switch to using Armbian-org GH hosted git shallow trees and Docker images

rpardini committed:
rpardini: armbian-next - slides - 7/x

rpardini committed:
armbian-next: pip3: workaround new problems introduced by pip 23+

rpardini committed:
armbian-next: use new/definitive pkg/image coordinates for Docker image, on Armbian org

rpardini committed:
armbian-next: artifact rootfs: change default OCI target to new Armbian org hosted repo

igorpecovnik committed:
Merge branch 'armbian-next'

igorpecovnik committed:
Armbian next CI/pre switch adjustements

- bump version to 23.05
- adjust OCI repositories, changed hash lenght to 12 and removed rootfs- prefix to shorten names (costmetic adjustement)
- build configs were moved to https://github.com/armbian/os
- remove CI that is not going to work anymore. This means also master is out of this CI
- small changed to Readme.txt. Just adding some note, removing obvious
- add propery - supported arhitectures for distribution. Needed for CI optimisation
- remove README.armbian-next.md (need to be somewhere else)
- shrink CODEOWNERS due to messages overload

Signed-off-by: Igor <igor@armbian.com>

igorpecovnik committed:
Merge branch 'main' into AR-1543

igorpecovnik committed:
Merge pull request #4847 from armbian/AR-1543

Update RK35xx legacy kernel to 5.10

dependabot[bot] committed:
build(deps): bump actions/upload-artifact from 3.0.0 to 3.1.2

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 3.0.0 to 3.1.2.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/6673cd052c4cd6fcf4b4e6e60ea986c889389535...0b7f8abb1508181956e8e162db84b466c27e18ce)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

igorpecovnik committed:
Merge pull request #4844 from armbian/dependabot/github_actions/actions/upload-artifact-3.1.2

build(deps): bump actions/upload-artifact from 3.0.0 to 3.1.2

hzyitc committed:
meson: bump current to v6.1 and edge to v6.2 (#4843)

* meson: bump current to v6.1 and edge to v6.2

adeepn committed:
JetHub: update preinstalled python packages to install via apt

rpardini committed:
git-ref2info: fix https Makefile URL for gitlab.com (eg, rk35xx-legacy) (#4848)

rpardini committed:
shellfmt/.editorconfig: enable trim whitespaces, run shellfmt, no actual code changes (#4849)

- shellfmt: bump to 3.6.0 (run with `bash lib/tools/shellfmt.sh`)
- .editorconfig: rationalize, remove duplicated logic

amazingfate committed:
remove u-boot patch fix-vcc3v3.patch for radxa-e25 (#4850)

adeepn committed:
Merge pull request #4852 from jethome-iot/AR-1545-update-jethub-prepackages-python-modules-to-install-via-apt

JetHub: update preinstalled python packages to install via apt

rpardini committed:
logs: remove `git diff` from `GIT_INFO_ANSI`, it's somewhat useless and potentially huge leading to `argument list too long`

rpardini committed:
kernel-config: ignore artifact cache, makes no sense with make menuconfig (fixes #4854) (#4856)

amazingfate committed:
use chroot_sdcard_apt_get_install for packages

amazingfate committed:
add shebang to rootfs-create.sh

amazingfate committed:
Merge pull request #4860 from amazingfate/chroot-apt-install

use chroot_sdcard_apt_get_install for packages

150balbes committed:
media: edge: move edge media to 6.2 [cherry-pick]

amazingfate committed:
not rsync /boot when copying root partition (#4859)

rpardini committed:
odroidxu4: fix the `custom_kernel_config__hack_odroidxu4_firmware()` for version hash (#4864)

belegdol committed:
Update odroidxu4-current to 5.4.232 (#4858)

igorpecovnik committed:
Rename config file "only_for" into "architecture" (#4861)

* Rename only_for def into architectures
* Rename def with typo arhitectures -> architectures

rpardini committed:
logging: introduce `TEXT_IS_TOO_DARK=yes` to make tool log output visible (temporary workaround)

- I am "almost" giving up on the "bright black" == "gray" thing. There's enough bad default themes out there, and not the first time people (rightfully) complained...

rpardini committed:
`uefi` / `rk3588-odroid` / `rockchip-rk3588`: `edge`: bump from `6.2-rc7` tag to `branch:linux-6.2.y`

rpardini committed:
docker: fix: pass down `OCI_TARGET_BASE` to Docker instance automatically

rpardini committed:
rpardini: configs: default to my OCI_TARGET_BASE

rpardini committed:
distro-agnostic.sh: fix: check if chroot (not host!) has `timesyncd.conf` as condition to enable `systemd-timesyncd.service` (#4870)

The-going committed:
sunxi: remove unused KERNELSOURCE="https://github.com/megous/linux"

The-going committed:
sunxi-6.2: add megous patches

The-going committed:
sunxi-6.2: add series.conf

igorpecovnik committed:
Update README.md

Closing https://github.com/armbian/build/issues/4871

EvilOlaf committed:
adjust wording (#4878)

adeepn committed:
Fix "No space left on device" while rsync root files to image

adeepn committed:
Merge pull request #4880 from jethome-iot/fix_create_size

Fix "No space left on device" while rsync root files to image

adeepn committed:
Replace absolute path to git-tree with relative, so it is accesible without docker builder.

adeepn committed:
Update meson64 edge branch to linux-6.2.y (#4876)

Reallign patches

The-going committed:
sunxi-6.2: add armbian patches

The-going committed:
sunxi-6.2: mfd: sunxi-ac200: fix error initialization

rpardini committed:
meson64: edge: rebase patches onto `v6.2.1`; fix mbox for recently added patch (no actual changes)

rpardini committed:
sunxi-6.2: rpardini's hacks / hammer to make work

rpardini committed:
bring back targets.conf and friends, for development?
